powershell/resources/psruntime/BuildTime/Cmdlets/ExportPsd1.cs

```diff
@@ -44,11 +44,24 @@ protected override void ProcessRecord()
           throw new ArgumentException($"Custom folder '{CustomFolder}' does not exist");
         }
 
+        string version = Convert.ToString(@"${$project.moduleVersion}");
+        string previewVersion = null;
+        if (version.Contains('-'))
+        {
+            string[] versions = version.Split('-');
+            if (versions.Length != 2)
+            {
+                throw new ArgumentException("Module-version is illegal");
+            }
+            version = versions[0];
+            previewVersion = versions[1];
+        }
+
```

Comment:
@dolauli , seems that the below items are not addressed:

If the first-part cannot be parsed into a windows version (i.e. x.y.z or x.y.z.w) then throw an error that the version is illegal
if the second-part is null or whitespace, throw an error that the version is illegal

and how about this one: In the NuSpec file, simply transmit the version as is to the file.

Reply:
@erich-wang A neat and more reasonable solution here is to validate whether 'moduleVersion' is a valid semantic version. But instead of doing here, it should be done in the autorest core, since the version configuration is not specific for powershell but for all the generators. Suggest file an issue to Garret to handle this.